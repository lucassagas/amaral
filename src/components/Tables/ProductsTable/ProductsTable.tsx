import {
  Button,
  HStack,
  Icon,
  Table,
  Tbody,
  Td,
  Text,
  Th,
  Thead,
  Tr,
} from '@chakra-ui/react'
import { Product } from '@prisma/client'
import NextLink from 'next/link'
import { BiTrashAlt } from 'react-icons/bi'
import { RiPencilLine } from 'react-icons/ri'

type ProductsTableProps = {
  products?: Product[]
}

export function ProductsTable({ products }: ProductsTableProps) {
  return (
    <Table colorScheme="whiteAlpha" mb="4">
      <Thead>
        <Tr>
          <Th>Nome</Th>
          <Th>
            <Text align="center">Quantidade</Text>
          </Th>
          <Th width="8"></Th>
        </Tr>
      </Thead>
      <Tbody>
        {products?.map((product) => (
          <Tr key={product.id}>
            <Td>
              <Text>{product.name}</Text>
            </Td>
            <Td>
              <Text align="center">{product.quantity}</Text>
            </Td>
            <Td p="0">
              <HStack>
                <NextLink href={`/app/products/update/${product.id}`} passHref>
                  <Button
                    cursor="pointer"
                    as="a"
                    size="sm"
                    fontSize="sm"
                    color="white"
                    colorScheme="orange"
                    pr="1"
                    leftIcon={<Icon as={RiPencilLine} fontSize="16" />}
                  />
                </NextLink>
                <Button
                  cursor="pointer"
                  as="a"
                  size="sm"
                  fontSize="sm"
                  color="white"
                  colorScheme="red"
                  pr="1"
                  leftIcon={<Icon as={BiTrashAlt} fontSize="16" />}
                />
              </HStack>
            </Td>
          </Tr>
        ))}
      </Tbody>
    </Table>
  )
}
