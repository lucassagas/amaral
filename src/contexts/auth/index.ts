export { AuthContext, AuthContextProvider, handleSignOut } from './auth'
