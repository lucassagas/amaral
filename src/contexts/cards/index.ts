export { CardContext, CardContextProvider } from './cards';

