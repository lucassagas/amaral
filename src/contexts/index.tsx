import { theme } from '@/styles/theme'
import { ChakraProvider } from '@chakra-ui/react'
import { ReactNode } from 'react'
import { QueryClientProvider } from 'react-query'
import { ReactQueryDevtools } from 'react-query/devtools'
import { queryClient } from 'src/lib/queryClient'
import { AuthContextProvider } from './auth'

type AppProviderProps = {
  children: ReactNode
}

export function AppProvider({ children }: AppProviderProps) {
  return (
    <ChakraProvider theme={theme}>
      <AuthContextProvider>
        <QueryClientProvider client={queryClient}>
          {children}
          <ReactQueryDevtools />
        </QueryClientProvider>
      </AuthContextProvider>
    </ChakraProvider>
  )
}
