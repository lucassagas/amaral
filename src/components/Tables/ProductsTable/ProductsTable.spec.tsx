import { AppProvider } from '@/contexts/index'
import { useBreakpointValue } from '@chakra-ui/react'
import { render, screen } from '@testing-library/react'
import { ProductsTable } from './ProductsTable'

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // Deprecated
    removeListener: jest.fn(), // Deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
})

jest.mock('@chakra-ui/react', () => ({
  ...jest.requireActual('@chakra-ui/react'),
  useBreakpointValue: jest.fn(),
}))

const products = [
  {
    id: '123',
    name: 'John Doe',
    quantity: 100,
    createdAt: new Date(),
    updatedAt: new Date(),
  },
]

const renderComponent = () => {
  return render(
    <AppProvider>
      <ProductsTable products={products} />
    </AppProvider>
  )
}

describe('<PageWrapper />', () => {
  it('should render correctly', () => {
    const { container } = renderComponent()

    expect(container).toMatchSnapshot()
  })

  it('should hidden update button when mobile screen', () => {
    ;(useBreakpointValue as jest.Mock).mockReturnValue({
      base: false,
      lg: true,
    })
    renderComponent()

    expect(screen.getByText(/editar/i)).toBeInTheDocument()
  })
})
