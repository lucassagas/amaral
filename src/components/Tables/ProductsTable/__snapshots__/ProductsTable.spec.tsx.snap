// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageWrapper /> should render correctly 1`] = `
<div>
  <table
    class="chakra-table css-e4kp03"
    role="table"
  >
    <thead
      class="css-0"
    >
      <tr
        class="css-0"
        role="row"
      >
        <th
          class="css-16juaxl"
        >
          <label
            class="chakra-checkbox css-1uiwwan"
          >
            <input
              class="chakra-checkbox__input"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
              value=""
            />
            <span
              aria-hidden="true"
              class="chakra-checkbox__control css-qlgo61"
            />
          </label>
        </th>
        <th
          class="css-1ednkar"
        >
          Nome
        </th>
        <th
          class="css-1ednkar"
        >
          <p
            class="chakra-text css-xi606m"
          >
            Quantidade
          </p>
        </th>
        <th
          class="css-ai8wzg"
        />
      </tr>
    </thead>
    <tbody
      class="css-0"
    >
      <tr
        class="css-0"
        role="row"
      >
        <td
          class="css-dl2l4m"
          role="gridcell"
        >
          <label
            class="chakra-checkbox css-1uiwwan"
          >
            <input
              class="chakra-checkbox__input"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
              value=""
            />
            <span
              aria-hidden="true"
              class="chakra-checkbox__control css-qlgo61"
            />
          </label>
        </td>
        <td
          class="css-1n48h1c"
          role="gridcell"
        >
          <p
            class="chakra-text css-0"
          >
            John Doe
          </p>
        </td>
        <td
          class="css-1n48h1c"
          role="gridcell"
        >
          <p
            class="chakra-text css-xi606m"
          >
            100
          </p>
        </td>
        <td
          class="css-1k2clil"
          role="gridcell"
        >
          <a
            class="chakra-button css-406sl6"
          >
            <span
              class="chakra-button__icon css-1wh2kri"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-ky7im0"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M15.728 9.686l-1.414-1.414L5 17.586V19h1.414l9.314-9.314zm1.414-1.414l1.414-1.414-1.414-1.414-1.414 1.414 1.414 1.414zM7.242 21H3v-4.243L16.435 3.322a1 1 0 0 1 1.414 0l2.829 2.829a1 1 0 0 1 0 1.414L7.243 21z"
                  />
                </g>
              </svg>
            </span>
            
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
